pkg/router/controller/factory/factory.go

```diff
@@ -25,30 +25,32 @@ import (
 // controller. It supports optional scoping on Namespace, Labels, and Fields of routes.
 // If Namespace is empty, it means "all namespaces".
 type RouterControllerFactory struct {
-	KClient        kclient.EndpointsNamespacer
-	OSClient       osclient.RoutesNamespacer
-	Namespaces     controller.NamespaceLister
-	ResyncInterval time.Duration
-	Namespace      string
-	Labels         labels.Selector
-	Fields         fields.Selector
+	EndpointsNamespacer kclient.EndpointsNamespacer
```

Comment:
In the long run we have to load services (for ingress).  In the short run I'd really prefer not to.  If we have the endpoints controller copy the idle annotation to endpoints, would that be sufficient to know here?